Add overview to docs (#713)
Qiskit/qiskit-nature: docs/index.rst

######################
Qiskit Nature overview
######################

Overview
==============

> **Qiskit Nature** is an open-source framework that supports problems including ground state energy computations,
> excited states and dipole moments of molecule, both open and closed-shell.

Reviewer: ```suggestion
**Qiskit Nature** is an open-source framework which supports solving natural science problems
using quantum computing algorithms. This includes finding ground and excited states of electronic
and vibrational structure problems, measuring the dipole moments of molecular systems, solving
the Ising and Fermi-Hubbard models on lattices, and much more.
```

The code comprises chemistry drivers, which when provided with a molecular

> configuration will return one and two-body integrals as well as other data that is efficiently

Reviewer: ```suggestion
configuration will return one- and two-body integrals as well as other data that is efficiently
```

computed classically. This output data from a driver can then be used as input in Qiskit

> Nature that contains logic which is able to translate this into a form that is suitable

Reviewer: I'm not yet too happy with the phrasing of this sentence but also don't have an immediate suggestion on how to improve it.. :thinking:

Reviewer: How about something like the following:
> Qiskit Nature provides logic to interpret this output data, translating it into a form that is suitable for quantum algorithms.

> for quantum algorithms. The conversion first creates a FermionicOperator which must then be mapped,
> e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.

Reviewer: ```suggestion
for quantum algorithms. The conversion first creates a fermionic operator which must then be mapped,
e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.
```

This aligns better with the qubit operator mentioned later. If we don't want this, we should say `FermionicOp`, matching the current name of the class.

######################
Qiskit Nature overview
######################

Overview
==============

> **Qiskit Nature** is an open-source framework which supports solving natural science problems

Reviewer: ```suggestion
**Qiskit Nature** is an open-source framework which supports solving quantum mechanical natural science problems
```

using quantum computing algorithms. This includes finding ground and excited states of electronic
and vibrational structure problems, measuring the dipole moments of molecular systems, solving
the Ising and Fermi-Hubbard models on lattices, and much more.

The code comprises chemistry drivers, which when provided with a molecular
configuration will return one and two-body integrals as well as other data that is efficiently
computed classically. This output data from a driver can then be used as input in Qiskit
Nature that contains logic which is able to translate this into a form that is suitable
for quantum algorithms. The conversion first creates a fermionic operator which must then be mapped,

> e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.

Reviewer: I would rephrase this similar to the following:
> For the solution of electronic structure problems, the problem Hamiltonian is first expressed in the second quantization formalism, comprising fermionic excitation and annihilation operators. These can then be mapped to the qubit formalism using a variety of mappings such as Jordan-Wigner, Parity, and more, in readiness for the quantum computation.

> 

Reviewer: Now we should add a section on vibrational structure calculations, which can also result from the data obtained by a classical driver.

And finally, we need to mention the lattice problems, which do not derive from classically obtained data.
